plugins/tests/greatexpectations/test_schema.py

```diff
@@ -0,0 +1,355 @@
+                local_file_path="/tmp/test3.parquet",  # noqa: F722
+            ),
+        ]
+    ) -> int:
+        return dataframe.shape[0]
+
+    @workflow
+    def valid_wf(dataframe: FlyteSchema) -> int:
+        return my_task(dataframe=dataframe)
+
+    df = pd.read_csv("data/yellow_tripdata_sample_2019-01.csv")
+    result = valid_wf(dataframe=df)
+    assert result == 10000
+
+
+def test_ge_schema_flyteschema_literal():
+    @task
+    def my_task(
+        dataframe: GreatExpectationsType[
+            FlyteSchema,
+            GreatExpectationsFlyteConfig(
+                datasource_name="data",
+                expectation_suite_name="test.demo",
+                data_connector_name="data_flytetype_data_connector",
+                batch_request_config=BatchRequestConfig(data_connector_query={"limit": 10}),
+                local_file_path="/tmp/test3.parquet",  # noqa: F722
+            ),
+        ]
+    ) -> int:
+        return dataframe.shape[0]
+
+    @workflow
+    def valid_wf() -> int:
+        df = pd.read_csv("data/yellow_tripdata_sample_2019-01.csv")
+        return my_task(dataframe=df)
+
+    result = valid_wf()
+    assert result == 10000
+
+
+def test_ge_schema_remote_flytefile():
+    ge_config = GreatExpectationsFlyteConfig(
+        datasource_name="data",
+        expectation_suite_name="test.demo",
+        data_connector_name="data_flytetype_data_connector",
+        local_file_path="/tmp",
+    )
+
+    @task
+    def my_task(dataset: GreatExpectationsType[FlyteFile[typing.TypeVar("csv")], ge_config]) -> int:
+        return len(pd.read_csv(dataset))
+
+    @workflow
+    def my_wf(dataset: FlyteFile[typing.TypeVar("csv")]) -> int:
+        return my_task(dataset=dataset)
+
+    result = my_wf(
+        dataset="https://raw.githubusercontent.com/superconductive/ge_tutorials/main/data/yellow_tripdata_sample_2019-01.csv"
+    )
+    assert result == 10000
+
+
+def test_ge_schema_remote_flytefile_literal():
+    ge_config = GreatExpectationsFlyteConfig(
+        datasource_name="data",
+        expectation_suite_name="test.demo",
+        data_connector_name="data_flytetype_data_connector",
+        local_file_path="/tmp",
+    )
+
+    @task
+    def my_task(dataset: GreatExpectationsType[FlyteFile[typing.TypeVar("csv")], ge_config]) -> int:
+        return len(pd.read_csv(dataset))
+
+    @workflow
+    def my_wf() -> int:
+        return my_task(
+            dataset="https://raw.githubusercontent.com/superconductive/ge_tutorials/main/data/yellow_tripdata_sample_2019-01.csv"
+        )
+
+    result = my_wf()
+    assert result == 10000
+
+
+def test_ge_local_flytefile():
+    ge_config = GreatExpectationsFlyteConfig(
+        datasource_name="data",
+        expectation_suite_name="test.demo",
+        data_connector_name="data_flytetype_data_connector",
+    )
+
+    @task
+    def my_task(dataset: GreatExpectationsType[FlyteFile[typing.TypeVar("csv")], ge_config]) -> int:
```

Comment:
can we add this as a predefined type here - https://github.com/flyteorg/flytekit/blob/master/flytekit/types/file/__init__.py#L31

Reply:
It's available now.